src/main/java/study/StringAddCalculator.java

```diff
@@ -0,0 +1,69 @@
+package study;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+    private static final String[] DEFAULT_DELIMITER = {",", ":"};
+    private static final int DEFAULT_FAILURE_VALUE = 0;
+    private static final Pattern PATTERN_FIND_CUSTOM_DELIMITER = Pattern.compile("//(.)\n(.+)");
+    private static final Pattern PATTERN_FIND_BODY_TEXT = Pattern.compile("(?://.\n)?(.+)");
+
+    private StringAddCalculator() {
+    }
+
+    public static int splitAndSum(String text) {
+        if (isEmpty(text)) {
+            return DEFAULT_FAILURE_VALUE;
+        }
+
+        String customDelimiter = getCustomDelimiter(text);
+        String[] numbers = splitWithCustomDelimiter(text, customDelimiter);
+        return getSumPositiveValue(numbers);
+    }
+
+    private static String[] splitWithCustomDelimiter(String text, String customDelimiter) {
+        String regex = String.join("|", DEFAULT_DELIMITER);
+        if (!isEmpty(customDelimiter)) {
+            regex += "|" + customDelimiter;
+        }
+        return getBodyText(text).split(regex);
+    }
+
+    private static String getCustomDelimiter(String text) {
+        return findFirstMatchedGroup(text, PATTERN_FIND_CUSTOM_DELIMITER);
+    }
+
+    private static String getBodyText(String text) {
+        return findFirstMatchedGroup(text, PATTERN_FIND_BODY_TEXT);
+    }
+
+    private static String findFirstMatchedGroup(String text, Pattern pattern) {
+        String result = "";
+        Matcher m = pattern.matcher(text);
+        if (m.find()) {
+            result = m.group(1);
+        }
+        return result;
+    }
+
+    private static int getSumPositiveValue(String[] numbers) {
+        int result = 0;
+        for (String number : numbers) {
+            result += parsePositiveInt(number);
+        }
+        return result;
+    }
+
+    private static int parsePositiveInt(String number) {
+        int value = Integer.parseInt(number);
+        if (Integer.signum(value) == -1) {
```

Comment:
```suggestion
        if (value < 0) {
```
이쪽이 좀 더 직관적이고 읽기 편하지 않을까요?